adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/impl/VertxBasedMqttProtocolAdapter.java

```diff
@@ -102,10 +102,16 @@ public void setMessageMapping(final MessageMapping<MqttContext> messageMappingSe
             final ResourceIdentifier targetAddress) {
 
         return getRegistrationAssertion(
-                targetAddress.getTenantId(),
-                targetAddress.getResourceId(),
-                ctx.authenticatedDevice(),
-                ctx.getTracingContext())
+                    targetAddress.getTenantId(),
+                    targetAddress.getResourceId(),
+                    ctx.authenticatedDevice(),
+                    ctx.getTracingContext())
+                .compose(handleAutoProvisioning(ctx,
+                    ctx.endpoint(),
+                    targetAddress.getTenantId(),
+                    targetAddress.getResourceId(),
+                    ctx.authenticatedDevice()
+                ))
```

Comment:
at this point we verify the registration status of the gateway itself. There is no need to handle auto provisioning as the device ID is the same as the authenticated (gateway) ID. In fact, the device ID is unknown at this point in time and will be (finally) established after the mapper has been invoked.
The auto provisioning will be handled in the *uploadMessage* method ...

Reply:
As auto-provisioning has been moved to the registration service completely, I guess this can be resolved?